packages/core-design-system/scripts/build.js

```diff
@@ -0,0 +1,287 @@
+    .sort()
+
+  // Organize files by type
+  const coreFiles = cssFiles.filter(file => !file.startsWith('dark-') && !file.startsWith('light-'))
+  const darkFiles = cssFiles.filter(file => file.includes('gitness') && file.startsWith('dark-'))
+  const lightFiles = cssFiles.filter(file => file.includes('gitness') && file.startsWith('light-'))
+
+  console.log('\n=== Theme File Summary (OSS) ===')
+  console.table({
+    'Dark Theme Files': { count: darkFiles.length },
+    'Light Theme Files': { count: lightFiles.length }
+  })
+  console.log('\n')
+
+  /**
+   * Core imports
+   * */
+  const coreStyles = `${getExportFileHeader()}
+   
+/* Core tokens */
+${coreFiles.map(file => `@import './${file}';`).join('\n')}`
+
+  /**
+   * OSS themes imports
+   * */
+  const ossContent = `${getExportFileHeader()}
+
+/* Theme files - Dark */
+${darkFiles.map(file => `@import './${file}';`).join('\n')}
+
+/* Theme files - Light */
+${lightFiles.map(file => `@import './${file}';`).join('\n')}`
+
+  /**
+   * Enterprise themes imports
+   * */
+  const enterpriseContent = `${getExportFileHeader()}
+
+/* Theme files - Dark */
+@import './dark-harness.css';
+
+/* Theme files - Light */
+@import './light-harness.css';`
+
+  // Write file
+  await fs.writeFile(`${DESIGN_SYSTEM_ROOT}/oss.css`, ossContent)
+  await fs.writeFile(`${DESIGN_SYSTEM_ROOT}/enterprise.css`, enterpriseContent)
+  await fs.writeFile(`${DESIGN_SYSTEM_ROOT}/core-imports.css`, coreStyles)
+
+  console.log('\n\x1b[1m\x1b[32m%s\x1b[0m', `✔︎ Created import files in ${DESIGN_SYSTEM_ROOT}`)
+}
+
+async function createEsmIndexFile() {
+  console.log(`\n\x1b[34mCreating ${DESIGN_SYSTEM_ROOT_ESM}/index.ts import file...\x1b[0m`)
+
+  // Get list of all JS files
+  const styleValueFiles = (await fs.readdir(DESIGN_SYSTEM_ROOT_ESM))
+    .filter(file => file.endsWith('.ts') && file !== 'index.ts')
+    .sort()
+
+  // Organize files by type
+  const coreFiles = styleValueFiles.filter(file => !file.startsWith('dark-') && !file.startsWith('light-'))
+  const darkFiles = styleValueFiles.filter(file => file.startsWith('dark-'))
+  const lightFiles = styleValueFiles.filter(file => file.startsWith('light-'))
+
+  console.log('\n=== Theme File Summary (ts) ===')
+  console.table({
+    'Dark Theme Files': { count: darkFiles.length },
+    'Light Theme Files': { count: lightFiles.length }
+  })
+  console.log('\n')
+
+  // Generate content
+  const content = `/**
+ * Harness Design System
+ * DO NOT UPDATE IT MANUALLY
+ * Generated on ${new Date().toUTCString()}
+ */
+
+  /* Theme files - Combined */
+export const designSystemThemeMap = {
+${[...darkFiles, ...lightFiles]
+  .map(file => {
+    const name = file.replace('.ts', '')
+    return `'${name}': '${name}',`
+  })
+  .join('\n')}
+}
+
+/* Core tokens */
+${coreFiles
+  .map(file => {
+    const fileName = file.replace('.ts', '')
+    const name = fileName.replace(/-./g, x => x[1].toUpperCase())
```

Comment:
This does not handle cases where multiple dashes exist (`some-theme-dark.ts` → `someThemeDark`).

```
const name = fileName.split('-').map((part, i) => i === 0 ? part : part.charAt(0).toUpperCase() + part.slice(1)).join('');

Reply:
It is getting converted properly. Please check the example below. 

```ts
export { default as darkHighContrastDeuteranopiaHarness } from './dark-high-contrast-deuteranopia-harness';
```